modules/internal/ChapelBase.chpl

```diff
@@ -307,11 +307,31 @@ module ChapelBase {
   //
   // % on primitive types
   //
-  inline proc %(a: int(?w), b: int(w)) return __primitive("%", a, b);
-  inline proc %(a: uint(?w), b: uint(w)) return __primitive("%", a, b);
+  inline proc %(a: int(?w), b: int(w)) {
+    if (chpl_checkDivByZero) then
+      if b == 0 then
+        halt("Attempt to divide by zero");
```

Comment:
Since we can be precise about which operation, I'd be inclined not to follow Python's lead.  It's too bad that the wording I chose for divide-by-zero doesn't have a symmetrical wording for % (that I can see).  "attempt to modulate by zero"?!? :)